### PR TITLE
Add unit-tests for the `onPassword/onProgress` callbacks of the API

### PR DESCRIPTION
**Pull request by Snuffleupagus**
The standard viewer implicitly depends on the `onPassword` and `onProgress` callbacks, in order to open password protected PDF files, respectively to report file loading progress. We currently have no unit-tests for this functionality, which seems unfortunate; hence this patch.

_Please note:_ Rather than adding more unit-tests to `api_spec.js`, I slightly extended/reworked two existing tests. Specifically for the password test, this _does not_ really change what we actually test, just how the test is done.

### CODE REVIEW

File: test/unit/api_spec.js [hunk 1]
```diff
@@ -38,12 +38,31 @@ describe('api', function() {
...
     }, 20000);
   }
+
   describe('PDFJS', function() {
     describe('getDocument', function() {
       it('creates pdf doc from URL', function() {
-        var promise = PDFJS.getDocument(basicApiUrl);
-        waitsForPromiseResolved(promise, function(data) {
-          expect(data instanceof PDFDocumentProxy).toEqual(true);
+        var getDocumentPromise = PDFJS.getDocument(basicApiUrl);
```
Comment (yurydelendik) [inline test/unit/api_spec.js]: (no action requred by worth mentioning: getDocument returns loadingTask, which contains the promise; we probably need to review the code base to refactor variable names)

**Comment by yurydelendik** [comment]
Looks good. See if it's easy to address getDocumentPromise naming thing now.

**Comment by Snuffleupagus** [comment]
I've changed this patch to use `loadingTask` instead of `getDocumentPromise`.
(The part about more general reviewing/refactoring of variable names in the code-base seemed better suited for a follow-up patch though.)

/botio unittest

**Comment by yurydelendik** [comment]
Thank you for the patch.

**Pull request merged**